hoomd/hpmc/ComputeSDF.h

```diff
@@ -518,61 +550,123 @@ template<class Shape> void ComputeSDF<Shape>::countHistogramLinearSearch(uint64_
                                     float(h_diameter.data[j]),
                                     float(h_charge.data[j]));
                                 }
-                            // loop over bins; only going up to min_bin-1 since we only
-                            // want the _first_ overlap of particle i with its neighbors
-                            for (size_t bin_to_sample = 0; bin_to_sample < min_bin; bin_to_sample++)
+                            // do compressive and expansive perturbations in the same loop, keeping
+                            // track of which overlap checks we do not need to perform loop over
+                            // bins; only going up to min_bin-1 since we only want the _first_
+                            // overlap of particle i with its neighbors
+                            bool do_compression = true;
+                            bool do_expansion = true;
+                            for (size_t bin_to_sample = 0;
```

Comment:
This would be more readable (and a smaller diff) if split into two separate for loops instead of one for loop that conspicuously goes to the max(i) and manages the different paths with bool flags.